spec/integration/data_types_spec.lua

```diff
@@ -0,0 +1,101 @@
+require("busted.runner")()
+local driver = require("luasql.exasol")
+local config = require("config")
+
+config.configure_logging()
+
+describe("Exasol data types", function()
+    local env = nil
+    local connection = nil
+
+    before_each(function()
+        env = driver.exasol()
+        local connection_params = config.get_connection_params()
+        connection = assert(env:connect(connection_params.source_name, connection_params.user,
+                                        connection_params.password))
+    end)
+
+    after_each(function()
+        assert.is_true(connection:close())
+        assert.is_true(env:close())
+    end)
+
+    describe("converted to correct Lua type", function()
+        local function cast(expression, type)
+            return string.format("cast(%s as %s)", expression, type)
+        end
+        -- See [list of all Exasol data types]
+        -- (https://docs.exasol.com/db/latest/sql_references/data_types/datatypesoverview.htm)
+        local test_cases = {
+            --
+            -- Boolean
+            --
+            {expression = "true", expected_value = true, expected_type = "BOOLEAN"},
+            {expression = "false", expected_value = false, expected_type = "BOOLEAN"}, --
+            --
+            -- Numeric types
+            --
+            {expression = "42", expected_value = 42, expected_type = "DECIMAL"},
+            {expression = "3.141", expected_value = "3.141", expected_type = "DECIMAL"},
+            {expression = cast("100.123456", "DECIMAL(5,2)"), expected_value = "100.12", expected_type = "DECIMAL"},
+            {expression = cast("3.141", "DOUBLE PRECISION"), expected_value = 3.141, expected_type = "DOUBLE"}, --
+            --
+            -- Sting types
+            --
+            {expression = "'abc'", expected_value = "abc", expected_type = "CHAR"},
+            {expression = cast("'abc'", "CHAR(5)"), expected_value = "abc  ", expected_type = "CHAR"},
+            {expression = cast("'abc'", "VARCHAR(5)"), expected_value = "abc", expected_type = "VARCHAR"}, --
+            --
+            -- Date/time types
+            --
+            {expression = cast("'2022-05-31'", "DATE"), expected_value = "2022-05-31", expected_type = "DATE"}, {
```

Comment:
Placement of opening curly bracket looks strange. We might need to check the formatter settings again.